epochX/cudacpp/CODEGEN/PLUGIN/CUDACPP_SA_OUTPUT/model_handling.py

```diff
@@ -911,12 +913,42 @@ def super_generate_parameters_class_files(self):
+            # For simplicity, compute these parameters directly from G, rather than from another such parameter
+            # (e.g. do not compute mdl_sqrt__aS as sqrt of aS, which would require defining aS first)
+            gparameters = { 'aS' : 'G * G / 4. / M_PI',
+                            'mdl_sqrt__aS' : 'G / 2. / constexpr_sqrt( M_PI )' }
+            gparamcoded = set()
             foundG = False
-            for line in self.write_hardcoded_parameters(self.params_dep).split('\n'):
-                if line != '':
-                    dcoupsetdpar.append( '    ' + line.replace('constexpr cxsmpl<double> mdl_G__exp__2','const fptype_sv mdl_G__exp__2').replace('constexpr double', 'const fptype_sv' if foundG else '//const fptype_sv' ) )
-                    if 'constexpr double G =' in line: foundG = True
-            replace_dict['dcoupsetdpar'] = '    ' + '\n'.join( dcoupsetdpar )
+            for pdep in self.params_dep:
+                ###misc.sprint(pdep.type, pdep.name) 
+                line = '    ' + self.write_hardcoded_parameters([pdep]).rstrip('\n')
+                ###misc.sprint(line)
+                if not foundG:
+                    # Comment out the default UFO assignment of mdl_sqrt__aS (from aS) and of G (from mdl_sqrt__aS), but keep them for reference
+                    # (WARNING! This Python CODEGEN code essentially assumes that this refers precisely and only to mdl_sqrt__aS and G)
+                    dcoupsetdpar.append( '    ' + line.replace('constexpr double', '//const fptype_sv') )
+                elif pdep.name == 'mdl_G__exp__2' : # bug fix: fptype, not double nor complex (UFO SUSY and SMEFT even disagree on this?)
+                    # Hardcode a custom assignment (valid for both SUSY and SMEFT) instead of replacing double or complex in 'line'
+                    dcoupsetdpar.append('        const fptype_sv ' + pdep.name + ' = G * G;' )
+                elif pdep.name == 'mdl_G__exp__3' : # bug fix: fptype, not double nor complex (UFO SUSY and SMEFT even disagree on this?)
+                    # Hardcode a custom assignment (valid for both SUSY and SMEFT) instead of replacing double or complex in 'line'
+                    dcoupsetdpar.append('        const fptype_sv ' + pdep.name + ' = G * G * G;' )
+                elif pdep.name in gparameters:
+                    # Skip the default UFO assignment from aS (if any?!) of aS and mdl_sqrt__aS, as these are now derived from G
+                    # (WARNING! no path to this statement! aS is not in params_dep, while mdl_sqrt__aS is handled in 'if not foundG' above)
+                    ###misc.sprint('Skip gparameter:', pdep.name)
+                    continue
+                else:
+                    for gpar in gparameters:
+                        if ' ' + gpar + ' ' in line and not gpar in gparamcoded:
+                            gparamcoded.add(gpar)
```

Comment:
more concerning the previous MR: this is more secure but not replacing the previous MR handling

Reply:
@oliviermattelaer  need to do something cleaner upstream to avoid those kind of "in"

Reply:
> @oliviermattelaer need to do something cleaner upstream to avoid those kind of "in"

Thanks Olivier. As discussed this is tracked here https://github.com/mg5amcnlo/mg5amcnlo/issues/90